driver-core/src/main/com/mongodb/internal/capi/MongoCryptOptionsHelper.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.capi;
+
+import com.mongodb.MongoClientException;
+import com.mongodb.crypt.capi.MongoAwsKmsProviderOptions;
+import com.mongodb.crypt.capi.MongoCryptOptions;
+import com.mongodb.crypt.capi.MongoLocalKmsProviderOptions;
+import org.bson.BsonDocument;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public final class MongoCryptOptionsHelper {
```

Comment:
I couldnt DRY up much code but I managed a tiny amount!

Reply:
It's something!